src/functions.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+namespace PhpFp;
+
+use PhpFp\Combinators\Curry;
+
+/**
+ * compose :: (b -> c), (a -> b) -> a -> c
+ */
+function compose(callable $f, callable $g): callable
+{
+    return static function ($x) use ($f, $g) {
+        return $f($g($x));
+    };
+}
+define('compose', '\PhpFp\compose');
```

Comment:
Are these `define`'s needed? They work, however the code looks a lot more confusing and the `use function` statements are a lot neater.

E.g.
```PHP
namespace Foo\Bar{
    function a() {
        return 'A';
    }

    define('a', '\Foo\Bar\a');
}

namespace Foo\Bass{

    echo a; // \Foo\Bar\a
    echo a(); // Uncaught Error: Call to undefined function Foo\Bass\a()
    echo (a)(); // A

    use function Foo\Bar\a;
    echo a(); // A
}
```
Or is there a good use case for these I have missed?